### PR TITLE
add namespace "radiation" around code related to radiation plugin

### LINKED ISSUE #1643: namespaces are missing in radiation plugin
Currently several methods in the radiation plugin are not hidden by a namespace. Some methods are not even included in the `namespace picongpu`. This should be fixed.

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request closes #1643. 

It surrounds the code related to the radiation plugin with the namespace `radiation`.

### CODE REVIEW

File: include/picongpu/param/radiationObserver.param [hunk 1]
```diff
@@ -124,5 +126,6 @@ namespace picongpu
 
     }
 
-  } // end namespace radiation_observer
-} // end namespace picongpu
+  } // namespace radiation_observer
```
Comment (n01r) [inline include/picongpu/param/radiationObserver.param]: We don't indent `namespace`s, just the code in the deepest nested `namespace` with 4 spaces (see [here](https://github.com/ComputationalRadiationPhysics/contributing/blob/master/codingGuideLines/cpp.md#6-namespaces)).

File: include/picongpu/param/radiation.param [hunk 2]
```diff
@@ -43,6 +43,8 @@
 #include "picongpu/particles/manipulators/manipulators.def"
 
 namespace picongpu
+{
+namespace radiation
 {
     namespace rad_linear_frequencies
```
Comment (sbastrakov) [inline include/picongpu/param/radiation.param]: Now when it is inside `picongpu::radiation`, the `rad` prefix is probably excessive (not only in this line).

File: include/picongpu/plugins/radiation/Radiation.kernel [hunk 3]
```diff
@@ -536,4 +538,6 @@ namespace picongpu
         }
     };
 
-}
+} // namepscae radiation
```
Comment (sbastrakov) [inline include/picongpu/plugins/radiation/Radiation.kernel]: Typo

**Comment by sbastrakov** [comment]
Looks good. However, now the plugin implementation files still do *not* follow the current namespace rules. For that, e.g. contents of `picongpu/include/picongpu/plugins/radiation/Radiation.kernel` should be in `picongpu::plugins::radiation` namespace, not just `picongpu::radiation` (with the latter being of course still an improvement).

File: include/picongpu/param/radiationObserver.param (hunk 1 above)
Comment (PrometheusPi) [reply include/picongpu/param/radiationObserver.param]: I addressed the extra indentation

**Comment by PrometheusPi** [comment]
@sbastrakov I added all your suggested improvements

**Comment by sbastrakov** [comment]
@PrometheusPi thanks!
Do you mind squashing some of the commits?

**Comment by PrometheusPi** [comment]
@sbastrakov I will squash it as soon as the compile test past successfully.

**Comment by PrometheusPi** [comment]
@sbastrakov I rebased the pull request and only kept the main changes as separate commits. Are you fine with this reduction?

**Pull request merged**